src/Shared/FileUtilities.cs

```diff
@@ -393,6 +398,48 @@ private static bool IsUNCPath(string path)
             return isUNC || path.IndexOf(@"\\?\globalroot", StringComparison.OrdinalIgnoreCase) != -1;
         }
 
+#if !CLR2COMPATIBILITY
+        /// <summary>
+        /// Gets the exact case used on the file system for an existing file or directory.
+        /// </summary>
+        /// <param name="path">A relative or absolute path.</param>
+        /// <returns>The full path using the correct case if the path exists. Otherwise, the full path.</returns>
+        private static string GetExactPath(string path)
```

Comment:
I'm a little worried about the perf impact here. There's some I/O here and some allocations.

Potentially this fix could be scoped down to only be used for normalizing project paths (~9 or so distinct locations) instead of **all** path normalization.

Reply:
Yeah, unfortunately I think that's worth doing.

Reply:
Considering how many times this method is called, it should be really scoped down. Maybe even scope it down to only apply on command line arguments like the issue suggests? If there's case mismatches in project files that run on case sensitive file systems, I'd be tempted to say it's actually user error.